### PR TITLE
Exception handling / it-academy.by / JC1-110-22 / Alena Trafimovich / Tasks1-7

### PR DESCRIPTION
**Pull request by AlenaTrafimovich**
## Pull request state

Actual status for PR:
- [x] implemented
- [x] reviewed
- [x] fixed
- [ ] approved

### CODE REVIEW

File: src/main/java/com/rakovets/course/java/core/practice/exception_handling/NPETask.java [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package com.rakovets.course.java.core.practice.exception_handling;
+
+public class NPETask {
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/exception_handling/NPETask.java:3]:
```suggestion
public class NpeTask {
```

File: src/main/java/com/rakovets/course/java/core/practice/exception_handling/NPEDemo.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
+package com.rakovets.course.java.core.practice.exception_handling;
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/exception_handling/NPEDemo.java:1]: move to `test` directory

File: src/test/java/com/rakovets/course/java/core/practice/exception_handling/NPETest.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
+package com.rakovets.course.java.core.practice.exception_handling;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+public class NPETest {
+    NPETask task = new NPETask();
+    NPEDemo demo = new NPEDemo();
```
Comment (rakovets) [inline src/test/java/com/rakovets/course/java/core/practice/exception_handling/NPETest.java:8]: remove it

File: src/test/java/com/rakovets/course/java/core/practice/exception_handling/NPETest.java [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+    @Test
+    public void testProduceNullPointerException() {
+        Assertions.assertThrows(NullPointerException.class, () -> task.produceNullPointerException());
+    }
+
+    @Test
+    public void testCatchNullPointerException() {
+        demo.catchNullPointerException();
+    }
```
Comment (rakovets) [inline src/test/java/com/rakovets/course/java/core/practice/exception_handling/NPETest.java:18]: remove it

File: src/test/java/com/rakovets/course/java/core/practice/exception_handling/NPETest.java [hunk 5]
```diff
@@ -0,0 +1,19 @@
+package com.rakovets.course.java.core.practice.exception_handling;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+public class NPETest {
+    NPETask task = new NPETask();
```
Comment (rakovets) [inline src/test/java/com/rakovets/course/java/core/practice/exception_handling/NPETest.java:7]: move to `@BeforeAll`

File: src/test/java/com/rakovets/course/java/core/practice/exception_handling/CustomArithmeticExceptionTest.java [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+public class CustomArithmeticExceptionTest {
+    CustomArithmeticExceptionTask task = new CustomArithmeticExceptionTask();
+    CustomArithmeticExceptionDemo demo = new CustomArithmeticExceptionDemo();
+
+    @Test
+    public void testProduceCustomArithmeticExceptionZero() {
+        Assertions.assertThrows(CustomArithmeticException.class, () -> task.produceCustomArithmeticException(2, 0));
```
Comment (rakovets) [inline src/test/java/com/rakovets/course/java/core/practice/exception_handling/CustomArithmeticExceptionTest.java:13]: catch a specific exception

File: src/test/java/com/rakovets/course/java/core/practice/exception_handling/CustomArithmeticExceptionTest.java [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+    }
+
+    @Test
+    public void testProduceCustomArithmeticException() {
+        Assertions.assertDoesNotThrow(() -> task.produceCustomArithmeticException(2, 2));
```
Comment (rakovets) [inline src/test/java/com/rakovets/course/java/core/practice/exception_handling/CustomArithmeticExceptionTest.java:18]: catch a specific exception

**Review by rakovets — CHANGES_REQUESTED** [review]
need to fix

File: src/test/java/com/rakovets/course/java/core/practice/exception_handling/CustomArithmeticExceptionTest.java (hunk 6 above)
Comment (AlenaTrafimovich) [reply src/test/java/com/rakovets/course/java/core/practice/exception_handling/CustomArithmeticExceptionTest.java:18]: I didn't understand exactly essence of correction. Could you explain,please

File: src/test/java/com/rakovets/course/java/core/practice/exception_handling/CustomArithmeticExceptionTest.java (hunk 7 above)
Comment (AlenaTrafimovich) [reply src/test/java/com/rakovets/course/java/core/practice/exception_handling/CustomArithmeticExceptionTest.java:23]: I didn't understand exactly essence of correction. Could you explain,please

**Review by rakovets — APPROVED** [review]

**Pull request closed without merging**